src/dataflow-types/types.rs

```diff
@@ -417,10 +428,12 @@ pub struct AvroEncoding {
     pub schema_registry_url: Option<Url>,
 }
 
-/// Encoding in CSV format, with no headers, and `n_cols` columns per row.
+/// Encoding in CSV format, with `n_cols` columns per row, with an optional header.
 #[serde(rename_all = "snake_case")]
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct CsvEncoding {
+    pub header_row: bool,
+    pub col_names: Vec<String>,
```

Comment:
I think making this an `Option<Vec<String>>` would be clearer than special-casing based on whether the vector is empty.

Reply:
Derp. Great catch.